### DISCUSSION TITLE
AssertionError: "At least one sample is shorter than the segment size."

### DISCUSSION THREAD

**Discussion by automationist** [coqui-ai/TTS#1082] [General Q&A]
I'm training on a custom dataset. This assertion error comes up and I'm unable to proceed with training. Could someone give any pointers on what I could be doing wrong?

**Comment by automationist** [comment]
If anyone encounters this issue, so far I've found that the `do_trim_silence=True` config option is the culprit. When I use `do_trim_silence=False`, it seems fine. But I'm not sure if that'd reduce the quality of the model. I'm gonna look into the `trim_silence` function to see what's up.

**Reply by Ca-ressemble-a-du-fake** [reply]
I have the same issue while fine tuning VITS model. Your workaround did not work, the error message remains. DId you find where the problem was located ?

It must come from on sample of VCTK, or Mailabs fr, or TTS-POrtuguese dataset because when they are not used the problem does not show. Maybe the [dataset analysis tool ](https://github.com/coqui-ai/TTS/blob/main/notebooks/dataset_analysis/AnalyzeDataset.ipynb)will point the culprit out!

**Comment by Ca-ressemble-a-du-fake** [comment]
To solve this issue I had to decrease spec_segment_size from 32 to 2.

**Reply by arif334** [reply]
> To solve this issue I had to decrease spec_segment_size from 32 to 2.

After doing this, I am getting another error:
```
RuntimeError: Argument #4: Padding size should be less than the corresponding input dimension, but got: padding (512, 512) at dimension 2 of input [1, 32, 512]
```
Any workaround?

**Comment by erogol** [comment]
I don't think the model would learn well with ```spec_segment_size from 32 to 2.``` 

Keep it 32 but remove shorter samples from your dataset. 

32 means ```32 * 256 / 22050``` secs in audio with a sampling rate of 22050

**Reply by stalevna** [reply]
I am trying to train vits model on ljspeech and getting this error: AssertionError:  [!] At least one sample is shorter than the segment size.
I have set do_trim_silence to False,
I have removed LJ045-0058, LJ018-0092,LJ001-0045, LJ002-0020  from .csv file after running  AnalyzeDataset notebook 
but I am still getting the error: AssertionError:  [!] At least one sample is shorter than the segment size.
The only thing that helps is setting spec_segment_size from 32 to 16, but I have no idea how it will affect the final result
my sampling rate = 16000
Could you help me figure out if it's ok to change the spec_segment_size to 16 or maybe there is something else I can do

**Comment by vince62s** [comment]
Are you sure ?
32 * 256 / 22050 = 0.37 seconds
those short segments are already filtered upfront (whether based on audio length or number of char / phonemes) so tis value does not make sense

**Reply by erogol** [reply]
it makes sense if you use silence trimming since filtering is done before trimming.

**Comment by yiwei0730** [comment]
I use the spec_segment_size from 32 to 16, and it can work,.
